File: Design Patterns Uni Project/SubChapter.cpp
#include "SubChapter.h"

void SubChapter::print()
{
	std::cout << "\tSubchapter: " << name << ": \n\n";
	for (Image & it : images)
	{
		it.print();
	}
	for (Paragraph & it : paragraphs)
	{
		it.print();
	}
	for (Table & it : tables)
	{
		it.print();
	}
}


void SubChapter::createNewParagraph(const std::string& newParagraph)
{
	Paragraph para(newParagraph);
	paragraphs.push_back(para);
}
void SubChapter::createNewImage(const std::string& newImage)
{
	Image img(newImage);
	images.push_back(img);
}
void SubChapter::createNewTable(const std::string& newTable)
{
	Table tbl(newTable);
	tables.push_back(tbl);
}

File: Design Patterns/src/Main/Models/SubChapter.java
package Main.Models;

import java.util.ArrayList;
import java.util.List;

public class SubChapter {
    private String name;
    private List<Element> elements = new ArrayList<>();

    public SubChapter(String name) {
        this.name = name;
    }

    public void createNewParagraph(String text) {
        Paragraph newParagraph = new Paragraph(text);
        elements.add(newParagraph);
    }

    public void createNewImage(String imageName) {
        Image newImage = new Image(imageName);
        elements.add(newImage);
    }

    public void createNewTable(String title) {
        Table newTable = new Table(title);
        elements.add(newTable);
    }

    public void print(){
        System.out.println("Subchapter: "+name);

        for(Element element:elements){
            element.print();
        }
    }
}
